### PR TITLE
[FIX] website_sale: fix viewed product add cart

### PR DESCRIPTION
**Pull request by fja-odoo**
If there is no product in the recently viewed product snippet it will be
hidden with a smooth transition.

Fix the last product that remained in snippet even if added to cart.

task-2076271

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale/static/src/js/website_sale_recently_viewed.js [hunk 1]
```diff
@@ -113,9 +114,9 @@ publicWidget.registry.productsRecentlyViewedSnippet = publicWidget.Widget.extend
         }).then(function (data) {
             wSaleUtils.updateCartNavBar(data);
             var $navButton = wSaleUtils.getNavBarButton('.o_wsale_my_cart');
-            wSaleUtils.animateClone($navButton, $(ev.currentTarget).parents('.o_carousel_product_card'), 25, 40);
-
-            self._dp.add(self._fetch()).then(self._render.bind(self));
+            wSaleUtils.animateClone($navButton, $(ev.currentTarget).parents('.o_carousel_product_card'), 25, 40).then(function () {
+                self._dp.add(self._fetch()).then(self._render.bind(self));
```
Comment (qsm-odoo) [inline addons/website_sale/static/src/js/website_sale_recently_viewed.js]: Do fetch before starting animation

File: addons/website_sale/static/src/js/website_sale_utils.js [hunk 2]
```diff
@@ -20,43 +20,51 @@ function getNavBarButton(selector) {
...
  * @param {Object} data
  */
 function updateCartNavBar(data) {
-    var $q = $(".my_cart_quantity");
-    $q.parents('li:first').removeClass('d-none');
-    $q.html(data.cart_quantity).hide().fadeIn(600);
+    var $qtyNavBar = $(".my_cart_quantity");
+    _.each($qtyNavBar, function (qty) {
+        $(qty).parents('li:first').removeClass('d-none');
+        $(qty).html(data.cart_quantity).hide().fadeIn(600);
```
Comment (qsm-odoo) [inline addons/website_sale/static/src/js/website_sale_utils.js]: Do not build `$(qty)` twice

File: addons/website_sale/static/src/js/website_sale_recently_viewed.js [hunk 3]
```diff
@@ -56,31 +56,32 @@ publicWidget.registry.productsRecentlyViewedSnippet = publicWidget.Widget.extend
...
+            productsGroups: mobileProducts,
+        }));
+        this.webCarousel = $(qweb.render('website_sale.productsRecentlyViewed', {
+            uniqueId: this.uniqueId,
+            productFrame: 4,
+            productsGroups: webProducts,
+        }));
+        this._addCarousel();
+        if (!products || !products.length) {
+            this.$el.removeClass('pt24 pb24').animate({'min-height': '', 'height': ''}, {duration: 1000});
```
Comment (qsm-odoo) [inline addons/website_sale/static/src/js/website_sale_recently_viewed.js]: Removing the padding classes is useless I guess, just need overflow hidden.

File: addons/website_sale/static/src/js/website_sale_recently_viewed.js [hunk 4]
```diff
@@ -56,31 +56,32 @@ publicWidget.registry.productsRecentlyViewedSnippet = publicWidget.Widget.extend
             return;
         }
         var products = res['products'];
-        if (products && products.length) {
```
Comment (qsm-odoo) [inline addons/website_sale/static/src/js/website_sale_recently_viewed.js]: I would keep the condition this way and add an `else` for the hiding part

Comment (fja-odoo) [reply addons/website_sale/static/src/js/website_sale_recently_viewed.js]: I have to render it when there is no product. That was a bug, the last product stayed in the carousel when added to cart .

File: addons/website_sale/static/src/js/website_sale_recently_viewed.js (hunk 3 above)
Comment (fja-odoo) [reply addons/website_sale/static/src/js/website_sale_recently_viewed.js]: Overflow is nice but the padding stays :/

**Pull request closed without merging**